integration/spark/src/main/scala/org/apache/spark/sql/optimizer/CarbonOptimizer.scala

```diff
@@ -37,427 +36,461 @@ import org.carbondata.spark.CarbonFilters
 /**
  * Carbon Optimizer to add dictionary decoder.
  */
-class CarbonOptimizer(optimizer: Optimizer, conf: CatalystConf)
-  extends Optimizer with PredicateHelper {
+object CarbonOptimizer {
 
-  val batches = Nil
+  def optimizer(optimizer: Optimizer, conf: CarbonSQLConf, version: String): Optimizer = {
+    CodeGenerateFactory.getInstance().optimizerFactory.createOptimizer(optimizer, conf)
+  }
 
-  override def execute(plan: LogicalPlan): LogicalPlan = {
+  def execute(plan: LogicalPlan, optimizer: Optimizer): LogicalPlan = {
```

Comment:
This is used from generated code from `CodeGenerateFactory.scala`, so we cannot delete it.